run_on_off_experiment.py

```diff
@@ -0,0 +1,249 @@
+import argparse
+import os
+import random
+from collections import namedtuple
+import pickle
+
+import gym
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import torch
+from torch.utils.tensorboard import SummaryWriter
+
+from sac_on_off import SAC
+
+torch.manual_seed(19971124)
+np.random.seed(42)
+random.seed(101)
+
+mse_loss_function = torch.nn.MSELoss()
+
+torch.autograd.set_detect_anomaly(True)
+
+if torch.cuda.device_count() > 0:
+    print("RUNNING ON GPU")
+    DEVICE = torch.device('cuda')
+else:
+    print("RUNNING ON CPU")
+    DEVICE = torch.device('cpu')
+
+
+def generate_plot(episodes, expert_ratios, scores, colour_len, log_freq, agent_type):
+    d = {"Episode": episodes, "ExpertRatio": expert_ratios}
+    experiments = pd.DataFrame(data=d)
+    experiments.insert(1, "Score", scores, allow_duplicates=False)
+    plot = sns.lineplot(data=experiments, x="Episode", y="Score", hue="ExpertRatio", legend="full",
+                        palette=sns.color_palette("hls", colour_len))
+    plot.set_xticks(range(0, max(episodes), log_freq))
+    plot.set_title(f"Scores of {agent_type}")
+
+    sns.lineplot(x=[ep for ep in range(0, max(episodes) + log_freq, log_freq)],
+                 y=[0 for _ in range(0, max(episodes) + log_freq, log_freq)],
+                 color="black", linestyle="--")
+
+    sns.lineplot(x=[ep for ep in range(0, max(episodes) + log_freq, log_freq)],
+                 y=[200 for _ in range(0, max(episodes) + log_freq, log_freq)],
+                 color="black", linestyle="--")
+
+    _path = os.path.join("experiments", "figures")
+    if not os.path.isdir(_path):
+        os.makedirs(_path, exist_ok=False)
+    figname = f"{agent_type}.png"
+    objname = f"{agent_type}.pkl"
+    f = open(os.path.join(_path, objname), 'wb')
+    pickle.dump(plot, f)
+    path = os.path.join(_path, figname)
+    plot.figure.savefig(path)
+
+
+def main(episodes_off, episodes_on, agent_type, num_trials):
+    # Log output
+    exp_name = "sac_on_off"
+    logdir = os.path.join("experiments", "logs", exp_name, agent_type)
+    os.makedirs(logdir, exist_ok=True)
```

Comment:
im not sure how exactly this works... but if you do exist_ok, i feel like there is a risk of it just making a completely new directory and deleting the old content. Could you use os.path.isdir() to check if its already a directory and only make it if it doesnt already exist?